Prepare spykfunc

matz-e opened the pull request:
Add spykfunc. Best to not be squashed to make future rebases easier?

Fixes #48.

matz-e committed:
mvdtool: new package @ 1.4

matz-e committed:
py-jprops: new package @ 2.0.2

matz-e committed:
fixup! py-jprops: new package @ 1.0.2

matz-e committed:
py-pyarrow: new package

matz-e committed:
py-h5py: add ~mpi to dependencies.

matz-e committed:
py-py4j: add version 0.10.6

matz-e committed:
py-py4j: change to sha256.

matz-e committed:
py-hdfs: new package

matz-e committed:
py-progress: new package.

matz-e committed:
py-pyspark: new package

matz-e committed:
py-sparkmanager: new package

matz-e committed:
py-bb5: new package

matz-e committed:
spykfunc: new package

pramodk commented on var/spack/repos/builtin/packages/mvdtool/package.py:
```diff
@@ -0,0 +1,79 @@
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
```
auto-added part below this should be removed

pramodk commented on var/spack/repos/builtin/packages/py-bb5/package.py:
```diff
@@ -0,0 +1,51 @@
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
```
unused?

pramodk commented on var/spack/repos/builtin/packages/py-h5py/package.py:
```diff
@@ -53,7 +53,7 @@ class PyH5py(PythonPackage):
+    depends_on('hdf5@1.8.4:+hl~mpi', when='~mpi')
 
     # MPI dependencies
     depends_on('hdf5+mpi', when='+mpi')
```
this should be :  `depends_on('hdf5@1.8.4:+hl+mpi', when='+mpi')`

pramodk commented on var/spack/repos/builtin/packages/py-jprops/package.py:
```diff
@@ -0,0 +1,51 @@
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
```
same as above (remove template)

pramodk commented on var/spack/repos/builtin/packages/spykfunc/package.py:
```diff
@@ -0,0 +1,61 @@
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
```
unused?

pramodk reviewed (changes requested):
Overall looks good to me.

matz-e committed:
Address comments.

pramodk commented:
Read comments on upstream PR and realise obvious things I didn't notice :)

matz-e commented:
Yes, I should have removed the boilerplates right away :/

I'll address the comments first thing in the morning, and will push another fix commit. I guess we should just squash this in the end, otherwise there's too much commit shuffling involved.

matz-e committed:
arrow: add missing dependency for python build

matz-e committed:
py-bb5: keep .git when creating the cache archive

matz-e committed:
feature: add option to retain .vcs

Some packages require version control information to build, e.g., python
packages depending on `setuptools-scm`.  Spack will remove any VCS
directory when caching hashable stages like tags, this patch will
provide a parameter `clean=False` to the `version` function that will
retain `.git` etc.

Should fix #8746.

pramodk commented on var/spack/repos/builtin/packages/py-jprops/package.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class PyJprops(PythonPackage):
+    """FIXME: Put a proper description of your package here."""
```
I was trying to push the description `Java properties file parser for Python` but seems like don't have permission to update PR from your branch:

```
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To https://github.com/matz-e/spack.git
 ! [remote rejected]     HEAD -> prepare-spykfunc (permission denied)
error: failed to push some refs to 'https://github.com/matz-e/spack.git'
```

matz-e replied:
It's my repository… I'll pick that fix.

matz-e committed:
spark: only java 8 is officially supported recently

matz-e committed:
py-py4j: add version 0.10.7 for Spark 2.3.2

matz-e committed:
{,py-py}spark: add BBP-specific binaries.

matz-e committed:
spykfunc: produce a working module

matz-e commented:
@pramodk I consider this a working version that can go into our tree now. All the necessary tidbits have been added to run a cluster-setup like in nix. Further fixes may have to happen, but these can be done post-merge.

pramodk commented:
Please retest

pramodk commented:
Please retest
